lib/reverse_sentence.rb

```diff
@@ -1,6 +1,54 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: Linear, O(n), where n is the length of the array
+# Space complexity: Constant, 0(1)
+
+# reverse_sentence method
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return nil if my_sentence == nil
+  string_reverse(my_sentence) # reverses order of the whole sentence (letters and words)
+  reverse_words(my_sentence) # reverses the order of each word (back to their original direction)
+  return my_sentence
+end
+
+# string_reverse helper method
+def string_reverse(my_sentence)
```

Comment:
You don't really need `string_reverse` if you have `reverse`. Just call `reverse(my_sentence, 0, my_sentence.length - 1)`.